src/bench/bench.cpp

```diff
@@ -125,37 +50,29 @@ void benchmark::BenchRunner::RunAll(Printer& printer, uint64_t num_evals, double
             assert(witness_enabled);
         }
 
-        if (!std::regex_match(p.first, baseMatch, reFilter)) {
-             g_testing_setup = nullptr;
-            continue;
-        }
-
-        uint64_t num_iters = static_cast<uint64_t>(p.second.num_iters_for_one_second * scaling);
-        if (0 == num_iters) {
-            num_iters = 1;
+        Bench bench;
+        bench.name(p.first);
+        if (asymptote.empty()) {
+            p.second(bench);
+        } else {
+            for (auto n : asymptote) {
+                bench.complexityN(n);
+                p.second(bench);
+            }
+            std::cout << bench.complexityBigO() << std::endl;
         }
-        State state(p.first, num_evals, num_iters, printer);
-        if (!is_list_only) {
-            p.second.func(state);
-        }
-        printer.result(state);
+        benchmarkResults.push_back(bench.results().back());
         g_testing_setup = nullptr;
     }
 
-    printer.footer();
-}
-
-bool benchmark::State::UpdateTimer(const benchmark::time_point current_time)
-{
-    if (m_start_time != time_point()) {
-        std::chrono::duration<double> diff = current_time - m_start_time;
-        m_elapsed_results.push_back(diff.count() / m_num_iters);
-
-        if (m_elapsed_results.size() == m_num_evals) {
-            return false;
+    if (!benchmarkResults.empty()) {
+        // Generate legacy CSV data to "benchmarkresults.csv"
+        std::ofstream fout("benchmarkresults.csv");
```

Comment:
thought: would `benchmark_results.csv` be more consistent with the project file naming (I'm not sure and won't bikeshed further)

Reply:
maybe I should add an option like `-csv=<filename>` that enables writing the .csv to the given file, if the option is present. Currently I always write a "benchmarkresults.csv", which can be a bit annoying when not wanted